src/Umbraco.Web/Trees/ContentTreeController.cs

```diff
@@ -61,14 +61,12 @@ protected override TreeNode GetSingleTreeNode(IEntitySlim entity, string parentI
                 //Special check to see if it ia a container, if so then we'll hide children.
                 var isContainer = entity.IsContainer;   // && (queryStrings.Get("isDialog") != "true");
 
-                var hasChildren = ShouldRenderChildrenOfContainer(entity);
```

Comment:
@kjac  + @nul800sebastiaan 

I'm pretty apprehensive about removing this! This check performs quite a lot of checks and security checks and now we're just ignoring all of that. The last time this was changed almost all the tree security was broken. 

This needs to be tested and re-tested asap

Reply:
`ShouldRenderChildrenOfContainer` is still invoked in `GetTreeNodesInternal` to ensure that users with custom start nodes don't get stranded. In its current implementation it never acts as a "limiter" for child nodes based on security - this is handled within `PerformGetTreeNodes`. It does however always return `false` for container nodes, unless the current user's start node is set underneath the container. 

As such, `ShouldRenderChildrenOfContainer` explicitly tells the clientside that container nodes never have children (unless the current user's start node is set underneath the container), based on the assumption that the client doesn't want to render the container children in the tree view (due to the potential volume). This assumption works for tree structures, because they don't need to hint that there are children to a container node. But the pickers do - they have to show the right arrow for both normal and container nodes. Therefore the assumption doesn't work.

What I have done is basically letting the clientside know if any given node has children without assuming anything about the application of this information. From there it's up to the client to render in the most performant way - i.e. never fetching all children for a container node.

Long story short: I can't really see a security issue here. That may be due to my lack of history on the project. If you can outline the scenario that causes a problem, of course I'll be happy to work it out 😄